packages/http-client-java/generator/http-client-generator-test/src/main/java/tsptest/patch/PatchClient.java

```diff
@@ -346,4 +418,30 @@ public Fish createOrUpdateFish(Fish fish) {
         JsonMergePatchHelper.getFishAccessor().prepareModelForJsonMergePatch(fish, false);
         return createOrUpdateFishWithResponse(fishInBinaryData, requestOptions).getValue().toObject(Fish.class);
     }
+
+    /**
+     * The createOrUpdateSalmon operation.
+     * 
+     * @param fish The fish parameter.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the second level model in polymorphic multiple levels inheritance which contains references to other
+     * polymorphic instances.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Salmon createOrUpdateSalmon(Salmon fish) {
+        // Generated convenience method for createOrUpdateSalmonWithResponse
+        RequestOptions requestOptions = new RequestOptions();
+        JsonMergePatchHelper.getFishAccessor().prepareModelForJsonMergePatch(fish, true);
+        BinaryData fishInBinaryData = BinaryData.fromObject(fish);
+        // BinaryData.fromObject() will not fire serialization, use getLength() to fire serialization.
+        fishInBinaryData.getLength();
+        JsonMergePatchHelper.getFishAccessor().prepareModelForJsonMergePatch(fish, false);
```

Comment:
If possible, add a local test doing same as this
```
        # create a "Salmon fish" instance, set one of its property to null

JsonMergePatchHelper.getFishAccessor().prepareModelForJsonMergePatch(fish, true);
        BinaryData fishInBinaryData = BinaryData.fromObject(fish);
        fishInBinaryData.getLength();
        JsonMergePatchHelper.getFishAccessor().prepareModelForJsonMergePatch(fish, false);

        # verify that fishInBinaryData contains a "property": null
```

Reply:
I will add in a follow up pr. Thanks.